romancal/outlier_detection/utils.py

```diff
@@ -0,0 +1,314 @@
+    ngroups = len(indices_by_group)
+    example_model = None
+    median_wcs = resamp.output_wcs
+
+    with input_models:
+        for i, indices in enumerate(indices_by_group):
+
+            drizzled_model = resamp.resample_group(input_models, indices)
+
+            if save_intermediate_results:
+                # write the drizzled model to file
+                _fileio.save_drizzled(drizzled_model, make_output_path)
+
+            if i == 0:
+                input_shape = (ngroups,) + drizzled_model.data.shape
+                dtype = drizzled_model.data.dtype
+                computer = MedianComputer(input_shape, in_memory, buffer_size, dtype)
+                example_model = drizzled_model
+
+            weight_threshold = compute_weight_threshold(drizzled_model.weight, maskpt)
+            drizzled_model.data[drizzled_model.weight < weight_threshold] = np.nan
+            computer.append(drizzled_model.data, i)
+            del drizzled_model
+
+    # Perform median combination on set of drizzled mosaics
+    median_data = computer.evaluate()
+
+    if save_intermediate_results:
+        # drizzled model already contains asn_id
+        _fileio.save_median(
+            example_model,
+            median_data,
+            median_wcs,
+            partial(make_output_path, asn_id=None),
+        )
+
+    return median_data, median_wcs
+
+
+def _median_without_resampling(
+    input_models,
+    maskpt,
+    weight_type,
+    good_bits,
+    save_intermediate_results,
+    make_output_path,
+    buffer_size=None,
+):
+    """
+    Compute median of data from models in a library.
+
+    Parameters
+    ----------
+    input_models : ModelLibrary
+        The input datamodels.
+
+    maskpt : float
+        The weight threshold for masking out low weight pixels.
+
+    weight_type : str
+        The type of weighting to use when combining images. Options are:
+        'ivm' (inverse variance) or 'exptime' (exposure time).
+
+    good_bits : int
+        The bit values that are considered good when determining the
+        data quality of the input.
+
+    save_intermediate_results : bool
+        if True, save the models and median model to fits.
+
+    make_output_path : function
+        The functools.partial instance to pass to save_median. Must be
+        specified if save_intermediate_results is True. Default None.
+
+    buffer_size : int
+        The size of chunk in bytes that will be read into memory when computing the median.
+        This parameter has no effect if the input library has its on_disk attribute
+        set to False.
+    """
+    in_memory = not input_models._on_disk
+    ngroups = len(input_models)
+    example_model = None
+
+    with input_models:
+        for i in range(len(input_models)):
+
+            model = input_models.borrow(i)
+            wht = build_driz_weight(
+                model,
+                # FIXME this was hard-coded to "ivm"
+                weight_type="ivm",
```

Comment:
should this get fixed now, since `weight_type` is getting passed into this function?  Or are you saying that this may change the behavior of the function in a way that is beyond this PR scope?  If the second one, has a ticket been made to ensure this gets fixed soon?

Reply:
Good idea. I opened an issue: https://github.com/spacetelescope/romancal/issues/1428